### ISSUE TITLE
No `strict` candidate in the completion list in `zip`

### ISSUE THREAD

**Issue by Molkree** [microsoft/pylance-release#1912]
## Environment data

-   Language Server version: v2021.10.0
-   OS and version: Win10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10

## Expected behaviour

`strict` is being suggested in the completions list.

## Actual behaviour

![image](https://user-images.githubusercontent.com/12860284/136420738-e500f28f-fee7-457f-b60a-9b162de8eb22.png)

## Code Snippet / Additional information

Try typing `strict` inside `zip`.

```python
names = ["John", "Jess"]
ages = [20]
for name, age in zip(names, ages):
    pass

```

Originally noticed in #1909

**Comment by erictraut** [comment]
This will be addressed in the next release. The signature help logic was looking for an `__init__` method and always preferring it over `__new__`. The `zip` class defines no `__init__` method, so the `__init__` method for `object` was being used instead. The new logic prefers the `__new__` method if the only `__init__` that is found is inherited from `object`.

**Comment by heejaechang** [comment]
fixed in 2021.11.0
